### PR TITLE
Prevent camera exposures when filterwheel is moving

### PR DESCRIPTION
**Pull request by AnthonyHorton**
Addresses #777 by implementing an `is_moving` property for filterwheels and checking it in the `take_exposure()` method of the `Camera` classes.

### CODE REVIEW

File: pocs/filterwheel/sbig.py [hunk 1]
```diff
@@ -62,6 +62,14 @@ def position(self):
             self.logger.warning("Filter wheel position unknown, returning NaN")
         return status['position']
 
+    @property
+    def is_moving(self):
+        """ Is the filterwheel currently moving """
+        status = self._SBIGDriver.cfw_query(self._handle)
+        if status['status'] == 'UNKNOWN':
+            self.logger.warning("{} returned 'UNKNOWN' status".self)
```
Comment (wtgee) [inline pocs/filterwheel/sbig.py]:
```suggestion
            self.logger.warning("{} returned 'UNKNOWN' status".format(self))
```

**Comment by AnthonyHorton** [comment]
I do like the idea of an `is_ready` property, and am planning to implement it. I think I will do it as wrapper around other properties/methods though in order to allow more flexibility. Even for the simple case of filterwheels being 'ready' entails both not moving _and_ being at the correct position. For cameras it can be more than just not currently taking an exposure, e.g. image sensor temperature within tolerances of the target value, and cooler power not at maximum.

**Comment by wtgee** [comment]
> I do like the idea of an `is_ready` property, and am planning to implement it. I think I will do it as wrapper around other properties/methods though in order to allow more flexibility. Even for the simple case of filterwheels being 'ready' entails both not moving _and_ being at the correct position. For cameras it can be more than just not currently taking an exposure, e.g. image sensor temperature within tolerances of the target value, and cooler power not at maximum.

:+1: yeah, seems like the `is_ready` would be good to support all around.

**Comment by AnthonyHorton** [comment]
Yup. And it's a step towards the asynchronous, distributed, messaging based model, too.

**Pull request merged**